packages/gatsby/cache-dir/navigation.js

```diff
@@ -16,7 +16,20 @@ const redirectMap = redirects.reduce((map, redirect) => {
 }, {})
 
 function maybeRedirect(pathname) {
-  const redirect = redirectMap[pathname]
+  let redirect = redirectMap[pathname]
+  if (!redirect) {
+    const redirectPath = Object.keys(redirectMap).find(possibleRedirectFrom => {
+      const possibleRedirect = redirectMap[possibleRedirectFrom]
+      if (possibleRedirect.ignoreCase) {
+        return (
+          possibleRedirect.fromPath.toLowerCase() === pathname.toLowerCase()
+        )
+      }
+
+      return possibleRedirect.fromPath === pathname
+    })
+    redirect = redirectMap[redirectPath]
+  }
```

Comment:
Could we generate a redirectsIgnoreCaseMap to ignore the find. Cause I'm guessing big sites have 1000 of redirects.
```suggestion
  let redirect = redirectMap[pathname]
  if (!redirect && redirectsIgnoreCaseMap[pathname.toLowerCase()]) {
    redirect = redirectMap[pathname.toLowerCase()]
  }
```

Reply:
That is an excellent idea. Thank you!